### DISCUSSION TITLE
Add e2e tests for ASP.NET Core

### DISCUSSION THREAD

**Discussion by isaaccloos** [metalbear-co/mirrord#438] [Ideas]
# Preamble

I was browsing available tickets and discovered a handful of [e2e test requests](https://github.com/metalbear-co/mirrord/issues?q=is%3Aopen+is%3Aissue+label%3Atests) for yet-to-be-supported frameworks. Namely:

- Java
- Rails

>At the time of writing, we run our e2e test flows on test apps in Node and Python (Flask). We should expand coverage to include more frameworks (since different web frameworks occasionally use libc/syscalls in different ways). Please implement the existing tests for a basic Java web server.

# Background

My professional work experience is with [dotnet](https://dotnet.microsoft.com/). And while there isn't a ticket for extending support to this platform, if it is in the interest of this project I'd be happy to contribute.

Highly common workflows for dotnet web development involve[^1]
- [Kestrel](https://learn.microsoft.com/aspnet/core/fundamentals/servers/kestrel?view=aspnetcore-6.0) for cross platform development
- [IIS / IIS Express](https://www.iis.net/) for Windows development 

# Request

If a new ticket was created to extend e2e support to the servers I've specified I'd be happy to take it. 

[^1]: These server applications are generalized. The reality is more complex of course, but for the purpose of this discussion I think it's close enough to the truth.

**Comment by eyalb181** [comment]
Hey @IsaacCloos, it's definitely in the interest of the project. I've opened this [issue](https://github.com/metalbear-co/mirrord/issues/439) - feel free to assign it to yourself. We defer to your judgement re exactly what platforms within .NET to test. 

Thanks!

**Reply by isaaccloos** [reply]
Forgive me, but after following this [guide](https://docs.github.com/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users) I don't believe I have permission to self assign this ticket. Could you do that when you've got the time. 

Thanks!

**Reply by eyalb181** [reply]
Oh sorry my fault. I think you might need to comment on the issue for me to be able to assign it to you though.
